components/proxy/conf/Caddyfile

```diff
@@ -237,13 +237,56 @@ https://{$GITPOD_DOMAIN} {
 
 		# note: no compression, as that breaks streaming for headless logs
 
-		uri strip_prefix /api
 		reverse_proxy server.{$KUBE_NAMESPACE}.{$KUBE_DOMAIN}:3000 {
-			import upstream_headers
-			import upstream_connection
+			rewrite /feature-flags/slow-database
 
-			# required for smooth streaming of terminal logs
-			flush_interval -1
+			@slow {
+				header X-Gitpod-Slow-Database "true"
+			}
+
+			@fast {
+				header X-Gitpod-Slow-Database "false"
+			}
+
+			@headerNotPresent {
+				header !X-Gitpod-Slow-Database
+			}
+
+			handle_response @fast {
+				uri strip_prefix /api
+
+				reverse_proxy server.{$KUBE_NAMESPACE}.{$KUBE_DOMAIN}:3000 {
+					import upstream_headers
+					import upstream_connection
+
+					# required for smooth streaming of terminal logs
+					flush_interval -1
```

Comment:
I'd like to be able to replace the body of these `reverse_proxy` directives with a snippet to reduce duplication but ran into this Caddy bug with nested imports: https://github.com/caddyserver/caddy/issues/4914

Reply:
Will take a look at `forward_auth`. When I looked before it didn't seem to match our use case, but will look again 👀

Reply:
> Would it work do do something like:
> ...

This looks like it should work according the Caddy docs on [intercepting responses](https://caddyserver.com/docs/caddyfile/directives/reverse_proxy#intercepting-responses):

I tried a simplified version in this Gist:

https://gist.github.com/andrew-farries/a2ac60d51e6c812ded573720c1ffabf6

but the fallthrough case doesn't work. Caddy docs say:

> When a response handler is invoked, the response from the backend is not written to the client, and the configured handle_response route will be executed instead, and it is up to that route to write a response. If the route does not write a response, then request handling will continue with any handlers that are ordered after this reverse_proxy.

but despite the `@fast` and `@none` matchers not writing a response, `respond` directive after the `reverse_proxy` directive never gets run.

(You can try it for yourself with `caddy run --watch` and then `http :2015/api/foo`)

Reply:
Another alternative to reduce duplication here would be to have the `feature-flags/slow-database` endpoint always set the `X-Gitpod-Slow-Database` response header. It currently only does so on `200 OK` responses:

https://github.com/gitpod-io/gitpod/blob/c154c3262a03a6b78ec40f8d07ef4e862cc9a52c/components/server/src/feature-flag/featureflag-controller.ts#L24-L44

That way we could remove the `@headerNotPresent` response matcher.

Reply:
Changed to use `forward_auth`, `copy_headers` and request matchers as we discussed in sync today. 

This needs https://github.com/gitpod-io/gitpod/pull/14930 to land first to ensure the `api/feature-flags/slow-database` endpoint returns `200 OK` consistently.